src/avif.c

```diff
@@ -282,16 +306,25 @@ void avifImageAllocatePlanes(avifImage * image, avifPlanesFlags planes)
         if (!image->yuvPlanes[AVIF_CHAN_Y]) {
             image->yuvRowBytes[AVIF_CHAN_Y] = fullRowBytes;
             image->yuvPlanes[AVIF_CHAN_Y] = avifAlloc(fullSize);
+            if (!image->yuvPlanes[AVIF_CHAN_Y]) {
+                return AVIF_RESULT_OUT_OF_MEMORY;
+            }
         }
 
         if (image->yuvFormat != AVIF_PIXEL_FORMAT_YUV400) {
             if (!image->yuvPlanes[AVIF_CHAN_U]) {
                 image->yuvRowBytes[AVIF_CHAN_U] = uvRowBytes;
                 image->yuvPlanes[AVIF_CHAN_U] = avifAlloc(uvSize);
+                if (!image->yuvPlanes[AVIF_CHAN_U]) {
+                    return AVIF_RESULT_OUT_OF_MEMORY;
```

Comment:
IMPORTANT: We need to make sure that `avifImageFreePlanes()` can free such a partially-allocated image.

I didn't look at this closely, but it seems that one way to make this work would be to set image->imageOwnsYUVPlanes to AVIF_TRUE early. Another way would be to free the earlier planes if the allocation of a later plane fails.

Reply:
Done.

This is brittle anyway: if you pass an `avifImage` with some plane set to external memory (with `imageOwnsYUVPlanes = AVIF_FALSE`), `avifImageAllocatePlanes()` will not replace those pointers but set `imageOwnsYUVPlanes = AVIF_TRUE`, leading to memory freed twice (unrelated to allocation failures).